projects/chaos/inc/power_path.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+// Chaos Power Path monitoring module.
+// Requires gpio, gpio interrupts, adcs, soft timers and interrupts to be initialized.
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "gpio.h"
+#include "soft_timer.h"
+#include "status.h"
+
+// Conversion callbacks for ADCs.
+typedef uint16_t (*PowerPathConversionCallback)(uint16_t value);
```

Comment:
What is this callback used for? I'm a little confused as to what it returns. Would a context parameter and source ID be useful here?

Reply:
It's just a function pointer to a conversion function for the ADC readout. I can rename it if you like? There's no point in context since I will need to write a separate function for each current and voltage conversion based on testing since the values/mappings won't necessarily be consistent between cases.

Reply:
It seems like this is intended primarily for internal use? I would prefer at least some comments because it wasn't immediately clear why it existed.

Reply:
Done. Added comment and renamed to make it clearer.